tsconfig.json

```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    // Target latest version of ECMAScript.
+    "target": "esnext",
```

Comment:
Would setting target to be `esnext` be ok?

Reply:
hey @shawnkoon! great question and thanks for reminding me about this PR :)

i think i'm gonna call this PR WIP as i recently learned how to use `@babel/typescript` so that we won't have to `tsc` then `babel` when publishing. So i'm gonna implement that eventually.... unless you'd want to take a stab at it.

basically, `babel.config.js` would look like 

```js
module.exports = {
  presets: [
    [
      '@babel/env',
      {
        targets: {
          node: 'current', // assuming you aren't targeting browsers if you're using this library.
        },
        debug: false,
      },
    ],
    '@babel/typescript',
    [
      '@babel/preset-react',
      {
        development: process.env.BABEL_ENV === 'development',
      },
    ],
  ],
  plugins: [
    '@babel/transform-modules-commonjs',
    '@babel/proposal-object-rest-spread',
  ],
};
```

and then typescript config could target `esnext` since `@babel/env` would handle the work of transpiling to target runtime.

here's some background on this concept:

* https://iamturns.com/typescript-babel/
* https://devblogs.microsoft.com/typescript/typescript-and-babel-7/

what do you think about that approach?

Reply:
> Would setting target to be `esnext` be ok?

also to actually answer this question, no probably not ideal if you were actually using TSC to build your distributed modules. That would definitely prevent this from being used in some environments, though we're really only using it in node 10+ right now :)

Reply:
Oh i see, yeah babel's typescript support should be good 👍 It was just rare occurrence for me to see `esnext` and `react` project so got lil curious xD thanks for the quick response!

Reply:
yeah totally! this is an unconventional experiment :)

Reply:
definitely. this is an unconventional use case, for sure. JSX -> slack for node.js projects